File: Coursera-Machine-Learning/Python-Code/ex1/computeCost.py
import numpy as np

def computeCost(X, y, theta):
    # init some useful values
    m = len(y) # number of training examples
    
    J = 0
    error = X.dot(theta) - y    
    sqr_error = error**2
    sum_sqr_error = np.sum(sqr_error)
    J = (1 / (2*m)) * sum_sqr_error
    
    return J

File: Coursera-Machine-Learning/Matlab-Code/machine-learning-ex1/ex1/computeCost.m
function J = computeCost(X, y, theta)
%COMPUTECOST Compute cost for linear regression
%   J = COMPUTECOST(X, y, theta) computes the cost of using theta as the
%   parameter for linear regression to fit the data points in X and y

% Initialize some useful values
m = length(y); % number of training examples

% You need to return the following variables correctly 
J = 0;

% ====================== YOUR CODE HERE ======================
% Instructions: Compute the cost of a particular choice of theta
%               You should set J to the cost.


% error vector of all training examples
error = X * theta - y;

sqr_error = error.^2; % Other, error.*error
sum_sqr_error = sum(sqr_error);

J = (1 / (2*m)) * sum_sqr_error;


% =========================================================================

end
